tests/conftest.py

```diff
@@ -140,6 +142,21 @@ def localhost_cert_private_key_file(
         yield tmp
 
 
+@pytest.fixture(scope="session")
+def ssl_context(
+    cert_authority: trustme.CA, localhost_cert: trustme.LeafCert
+) -> ssl.SSLContext:
+    ssl_context = ssl.create_default_context()
+
+    if hypercorn is None:
+        ssl_context.load_verify_locations(certifi.where())
```

Comment:
create_default_context should already load the correct verify locations

Reply:
I don't know if we need `certifi` in *this particular case* or not, but just to help make this clearer for anyone following the conversation...

Python's `ssl.create_default_context()` doesn't hook into the system certificate store, so it's actually *not*  able to correctly validate https hostnames...

```python
>>> import ssl, socket
>>> context = ssl.create_default_context()
>>> sock = socket.create_connection(("www.example.com", 443))
>>> ssl_sock = context.wrap_socket(sock, server_hostname="www.example.com")
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 423, in wrap_socket
    session=session
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 870, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/ssl.py", line 1139, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1076)
```

Which is why certifi is needed (A third party package which bundles up Mozilla's certificate store of root certificates)...

```python
>>> import certifi, ssl, socket
>>> context = ssl.create_default_context()
>>> context.load_verify_locations(certifi.where())
>>> sock = socket.create_connection(("www.example.com", 443))
>>> ssl_sock = context.wrap_socket(sock, server_hostname="www.example.com")
```